core/store/ledgerstore/event_store.go

```diff
@@ -208,3 +210,124 @@ func (this *EventStore) getEventNotifyByTxKey(txHash common.Uint256) []byte {
 	copy(key[1:], data)
 	return key
 }
+
+func (this *EventStore) SaveContractMetaDataEvent(height uint32, metaData *payload.MetaDataCode) error {
+	key := getContractMetaDataKey(height, metaData.Contract)
+	value := common.NewZeroCopySink(64)
+	metaData.Serialization(value)
+	this.store.BatchPut(key, value.Bytes())
+	return nil
+}
+
+func (this *EventStore) GetContractMetaDataEvent(height uint32, contractAddr common.Address) (*payload.MetaDataCode, error) {
+	key := getContractMetaDataKey(height, contractAddr)
+	data, err := this.store.Get(key)
+	if err != nil {
+		return nil, err
+	}
+	source := common.NewZeroCopySource(data)
+	metaData := &payload.MetaDataCode{}
+	err = metaData.Deserialization(source)
+	if err != nil {
+		return nil, err
+	}
+	return metaData, nil
+}
+
+func getContractMetaDataKey(height uint32, contractAddr common.Address) []byte {
+	key := common.NewZeroCopySink(10)
+	key.WriteByte(byte(scom.CROSS_SHARD_CONTRACT_META))
+	key.WriteUint32(height)
+	key.WriteBytes(contractAddr[:])
+	return key.Bytes()
+}
+
+func (this *EventStore) SaveContractEvent(height uint32, deployCode *payload.DeployCode) error {
```

Comment:
there is no event to save, I think rename to `SaveDeployCode` should be more reasonable.

Reply:
done